### PR TITLE
Add px/redis_stats script.

### PR DESCRIPTION
**Pull request by htroisi**
This is a fork of the `px/psql_stats` script with the following modifications:
- Add a Request Throughput per Command timeseries chart

image

- Convert the "Redis IP" column to a pod name (if possible).

### CODE REVIEW

File: pxl_scripts/px/redis_stats/redis_stats.pxl [hunk 1]
```diff
@@ -0,0 +1,284 @@
...
+
+
+def latency_histogram(start: str, pod: px.Pod):
+    ''' Computes a histogram of HTTP request latency.
+
+    Args:
+    @start: The timestamp of data to start at.
+    @svc: the partial/full-name of the svc.
+
+    Returns: DataFrame of the HTTP latency histogram for svcs that
```
Comment (oazizi000) [inline pxl_scripts/px/redis_stats/redis_stats.pxl:89]: ditto

**Review by oazizi000 — APPROVED** [review]
We need to account for trace_role in these scripts, but that's a separate task. Let's move ahead with this.

File: pxl_scripts/px/redis_stats/redis_stats.pxl [hunk 2]
```diff
@@ -0,0 +1,284 @@
...
+def cmd_timeseries(start: str, pod: px.Pod):
+    ''' Calculate Redis cmd timeseries for redis traffic per Pod, database pair.
+
+    @start: The timestamp of data to start at.
+    @pod: The partial/full-name of the pod to monitor.
+
+    Returns: Returns the DataFrame containing Redis cmd time-series for redis
+        traffic for each database Pods talk to.
+    '''
+    df = px.DataFrame(table='redis_events.beta', start_time=start)
```
Comment (oazizi000) [inline pxl_scripts/px/redis_stats/redis_stats.pxl:116]: For consistency, add: 
```df = format_redis_table(df)```

File: pxl_scripts/px/redis_stats/redis_stats.pxl [hunk 3]
```diff
@@ -0,0 +1,284 @@
...
+    df.pod = df.ctx['pod']
+    df = df[px.contains(df.pod, pod)]
```
Comment (oazizi000) [inline pxl_scripts/px/redis_stats/redis_stats.pxl:118]: For consistency: ```df = df[px.contains(df[k8s_object], pod)]```

File: pxl_scripts/px/redis_stats/redis_stats.pxl [hunk 4]
```diff
@@ -0,0 +1,284 @@
...
+    df.timestamp = px.bin(df.time_, window_ns)
+    df.time_ = df.timestamp
```
Comment (oazizi000) [inline pxl_scripts/px/redis_stats/redis_stats.pxl:120]: Collapse into one line: df.time_ = px.bin(df.time_, window_ns)

**Review by oazizi000 — COMMENTED** [review]
Proposal: How about we limit the results to `trace_role == 2` for now, and then we can make the column names: "server" and "client".

This is more generic than "pod", which if the server is outside the cluster, is not a pod.

Then overtime, we can add `trace_role==1` results into the script, correctly putting the servers under server and the clients under client.

**Review by oazizi000 — APPROVED** [review]

**Review by zasgar — APPROVED** [review]

**Pull request merged**